scuba/config.py

```diff
@@ -155,9 +155,12 @@ def override(self, node: yaml.nodes.Node) -> OverrideMixin:
         assert isinstance(obj, OverrideMixin)
         return obj
 
+    def not_yet_implemented(self, node: yaml.nodes.Node):
+        pass
 
 Loader.add_constructor("!from_yaml", Loader.from_yaml)
 Loader.add_constructor("!override", Loader.override)
+Loader.add_constructor("!reference", Loader.not_yet_implemented)
```

Comment:
`!reference` is only added for compatibility with content imported from `.gitlab-ci.yml` (via `!from_yaml`), correct? It is not a part of the documented `.scuba.yml` schema.

So I don't think we should be adding this constructor to this `Loader` class. It should be part of a separate Loader class used only when parsing `.gitlab-ci.yml` files. And how do we know we're parsing a `.gitlab-ci.yml` file (or a differently-named file using its schema)? We should use a new tag + constructor which is a variant of `!from_yaml`.

This is all spelled out in #200.